### PR TITLE
Functional test - Add quick access test

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 1.7.8.x
| Description?      | Add quick access test
| Type?             | refacto
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | no
| How to test?      | TEST_PATH="functional/BO/15_header/*" URL_FO=yourShopURL npm run specific-test
| Possible impacts? | no

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/24699)

### CODE REVIEW

File: tests/UI/pages/commonPage.js [hunk 1]
```diff
@@ -212,9 +212,10 @@ class CommonPage {
    * @param accept {boolean} True to accept the dialog, false to dismiss
    * @return {Promise<void>}
    */
-  async dialogListener(page, accept = true) {
+  async dialogListener(page, accept = true, text = '') {
```
Comment (Progi1984) [inline tests/UI/pages/commonPage.js:216]: Please add JSDoc

**Review by Progi1984 — COMMENTED** [review]
A small comment

File: tests/UI/pages/commonPage.js (hunk 1 above)
Comment (nesrineabdmouleh) [reply tests/UI/pages/commonPage.js:216]: Done thx :+1:

File: tests/UI/pages/commonPage.js [hunk 2]
```diff
@@ -210,11 +210,13 @@ class CommonPage {
...
+   * @param text {string} Text to set on dialog input
    * @return {Promise<void>}
    */
-  async dialogListener(page, accept = true) {
+  async dialogListener(page, accept = true, text = '') {
     page.once('dialog', (dialog) => {
-      if (accept) dialog.accept();
+      if (accept && text === '') dialog.accept();
+      else if (text !== '') dialog.accept(text);
       else dialog.dismiss();
```
Comment (boubkerbribri) [inline tests/UI/pages/commonPage.js]:
```suggestion
      if (accept && text === '') {
        dialog.accept();
      }
      
      else if (text !== '') {
        dialog.accept(text);
      }
      
      else {
        dialog.dismiss();
      }
```

**Review by boubkerbribri — COMMENTED** [review]
Just a little suggestion to improve readability

File: tests/UI/pages/commonPage.js (hunk 2 above)
Comment (nesrineabdmouleh) [reply tests/UI/pages/commonPage.js]: Done thx :+1:

File: admin-dev/themes/default/template/header.tpl [hunk 3]
```diff
@@ -139,22 +139,22 @@
             <li class="divider"></li>
             {if isset($matchQuickLink)}
               <li>
-                <a href="javascript:void(0);" class="ajax-quick-link" data-method="remove"
+                <a id="quick_remove" href="javascript:void(0);" class="ajax-quick-link" data-method="remove"
```
Comment (boubkerbribri) [inline admin-dev/themes/default/template/header.tpl]:
```suggestion
                <a id="quick-remove-link" href="javascript:void(0);" class="ajax-quick-link" data-method="remove"
```

File: admin-dev/themes/default/template/header.tpl [hunk 4]
```diff
@@ -139,22 +139,22 @@
...
                   data-quicklink-id="{$matchQuickLink}">
                   <i class="material-icons">remove_circle</i>
                   {l s='Remove from QuickAccess' d='Admin.Navigation.Header'}
                 </a>
               </li>
             {else}
               <li>
-                <a href="javascript:void(0);" class="ajax-quick-link" data-method="add">
+                <a id="quick_add" href="javascript:void(0);" class="ajax-quick-link" data-method="add">
```
Comment (boubkerbribri) [inline admin-dev/themes/default/template/header.tpl]:
```suggestion
                <a id="quick-add-link" href="javascript:void(0);" class="ajax-quick-link" data-method="add">
```

File: admin-dev/themes/default/template/header.tpl [hunk 5]
```diff
@@ -139,22 +139,22 @@
...
                   <i class="material-icons">add_circle</i>
                   {l s='Add current page to QuickAccess' d='Admin.Navigation.Header'}
                 </a>
               </li>
             {/if}
             <li>
-              <a href="{$link->getAdminLink("AdminQuickAccesses")|addslashes}">
+              <a id="quick_manage" href="{$link->getAdminLink("AdminQuickAccesses")|addslashes}">
```
Comment (boubkerbribri) [inline admin-dev/themes/default/template/header.tpl]:
```suggestion
              <a id="quick-manage-link" href="{$link->getAdminLink("AdminQuickAccesses")|addslashes}">
```

File: admin-dev/themes/new-theme/template/components/layout/quick_access.tpl [hunk 6]
```diff
@@ -13,7 +13,7 @@
     {/foreach}
     <div class="dropdown-divider"></div>
     {if isset($matchQuickLink)}
-      <a
+      <a id="quick_remove"
```
Comment (boubkerbribri) [inline admin-dev/themes/new-theme/template/components/layout/quick_access.tpl]:
```suggestion
      <a id="quick-remove-link"
```

File: admin-dev/themes/new-theme/template/components/layout/quick_access.tpl [hunk 7]
```diff
@@ -29,7 +29,7 @@
         {l s='Remove from Quick Access' d='Admin.Navigation.Header'}
       </a>
     {else}
-      <a
+      <a id="quick_add"
```
Comment (boubkerbribri) [inline admin-dev/themes/new-theme/template/components/layout/quick_access.tpl]:
```suggestion
      <a id="quick-add-link"
```

File: admin-dev/themes/new-theme/template/components/layout/quick_access.tpl [hunk 8]
```diff
@@ -44,7 +44,7 @@
         {l s='Add current page to Quick Access'  d='Admin.Actions'}
       </a>
     {/if}
-    <a class="dropdown-item" href="{$link->getAdminLink("AdminQuickAccesses")|addslashes}">
+    <a id="quick_manage" class="dropdown-item" href="{$link->getAdminLink("AdminQuickAccesses")|addslashes}">
```
Comment (boubkerbribri) [inline admin-dev/themes/new-theme/template/components/layout/quick_access.tpl]:
```suggestion
    <a id="quick-manage-link" class="dropdown-item" href="{$link->getAdminLink("AdminQuickAccesses")|addslashes}">
```

File: tests/UI/pages/BO/BObasePage.js [hunk 9]
```diff
@@ -26,6 +26,14 @@ class BOBasePage extends CommonPage {
...
     this.shopVersionBloc = '#shop_version';
     this.headerShopNameLink = '#header_shopname';
+    this.headerQuickDiv = '#header_quick';
+    this.quickAccessDropdownToggle = '#quick_select';
+    this.quickAccessContainer = '#quick-access-container';
+    this.quickAccessLink = idLink => `${this.headerQuickDiv} li:nth-child(${idLink}) a,
+    ${this.quickAccessContainer} a:nth-child(${idLink})`;
+    this.quickAddCurrentLink = '#quick_add';
+    this.quickAccessRemoveLink = '#quick_remove';
+    this.manageYourQuickAccessLink = '#quick_manage';
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/BObasePage.js]:
```suggestion
    this.quickAddCurrentLink = '#quick-add-link';
    this.quickAccessRemoveLink = '#quick-remove-link';
    this.manageYourQuickAccessLink = '#quick-manage-link';
```

File: tests/UI/pages/BO/BObasePage.js [hunk 10]
```diff
@@ -26,6 +26,14 @@ class BOBasePage extends CommonPage {
     this.userProfileLogoutLink = 'a#header_logout';
     this.shopVersionBloc = '#shop_version';
     this.headerShopNameLink = '#header_shopname';
+    this.headerQuickDiv = '#header_quick';
+    this.quickAccessDropdownToggle = '#quick_select';
+    this.quickAccessContainer = '#quick-access-container';
+    this.quickAccessLink = idLink => `${this.headerQuickDiv} li:nth-child(${idLink}) a,
+    ${this.quickAccessContainer} a:nth-child(${idLink})`;
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/BObasePage.js]: You can add a class named `quick-row-link` to li (on legacy) and a (on migrated pages).
And use it like this : 

```suggestion
    this.quickAccessLink = idLink => `.quick-row-link:nth-child(${idLink}) `;
```

**Review by boubkerbribri — COMMENTED** [review]
Some improvement to match new core conventions for selectors

File: tests/UI/pages/BO/BObasePage.js (hunk 10 above)
Comment (nesrineabdmouleh) [reply tests/UI/pages/BO/BObasePage.js]: Done thx :+1:

**Review by Progi1984 — DISMISSED** [review]

File: tests/UI/pages/BO/BObasePage.js (hunk 10 above)
Comment (boubkerbribri) [reply tests/UI/pages/BO/BObasePage.js]: The `dropdown-item` class exist for other elements on the page and it's not specific to the quick access block, that's why adding another specific class for both legacies and migrated pages could be better

File: admin-dev/themes/default/template/header.tpl [hunk 11]
```diff
@@ -129,7 +129,7 @@
           <ul class="dropdown-menu">
             {if !empty($quick_access)}
               {foreach $quick_access as $quick}
-                <li {if $link->matchQuickLink({$quick.link})}{assign "matchQuickLink" $quick.id_quick_access}class="active"{/if}>
+                <li class="quick-row-link" {if $link->matchQuickLink({$quick.link})}{assign "matchQuickLink" $quick.id_quick_access}class="active"{/if}>
```
Comment (boubkerbribri) [inline admin-dev/themes/default/template/header.tpl]:
```suggestion
                <li class="quick-row-link{if $link->matchQuickLink({$quick.link})}{assign "matchQuickLink" $quick.id_quick_access} active{/if}">
```

**Review by boubkerbribri — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Comment by Progi1984** [comment]
Thanks @nesrineabdmouleh & @boubkerbribri

**Pull request merged**